script/reproducible/verify.sh

```diff
@@ -0,0 +1,195 @@
+  exit 0
+elif [[ "$OSTYPE" == "darwin"* ]]; then
+  # If you are on mac run youself pointing to the correct version of toolchain
+  setup_mac_for_dev "${CHECK_VERSION}" || true
+  echo "Executing under macOS image, continuing"
+fi
+
+export ELECTRON_OUT_DIR=Default
+export ELECTRON_ENABLE_STACK_DUMPING=1
+
+echo Check for disk space
+export FREESPACE=$(BLOCKSIZE=1g df $buildir | tail -n1 | awk '{print $4}')
+export MINSPACE=100 # number of gygabites needed to download and build electron
+
+if [ $FREESPACE -ge $MINSPACE ] || [ "$nodiskcheck" = "true" ]; then
+  echo yes enough free space
+else
+  echo not enough free space - ${MINSPACE}GB 
+  exit 5
+fi
+
+export GN_CONFIG=release
+export STRIP_BINARIES=true 
+export GENERATE_SYMBOLS=true
+export CHECK_DIST_MANIFEST=1
+
+echo "Building $env:GN_CONFIG build"
+git config --global core.longpaths true
+mkdir -p $buildir
+cd $buildir
+export CHROMIUM_BUILDTOOLS_PATH=$buildir/src/buildtools
+export SCCACHE_PATH=$buildir/src/electron/external_binaries/sccache
+#export GCLIENT_EXTRA_ARGS="$GCLIENT_EXTRA_ARGS --custom-var=\"checkout_boto=True\" --custom-var=\"checkout_requests=True\""
+gclient config --name "src/electron" --unmanaged $GCLIENT_EXTRA_ARGS "https://github.com/electron/electron"
+gclient sync -f --with_branch_heads --with_tags
+cd src
+cd electron
+git fetch
+git checkout $CHECK_VERSION
+gclient sync -f
+cd ..
+cd ..
+export BUILD_CONFIG_PATH="//electron/build/args/release.gn"
+# build out/Default
+$SCCACHE_PATH --show-stats
+cd src
+# XXX: leave extra space after GN_EXTRA_ARG
+# GN_EXTRA_ARG= 
+gn gen out/Default --args="import(\"//electron/build/args/release.gn\") $GN_EXTRA_ARG cc_wrapper=\"$SCCACHE_PATH\""
+ninja -C out/Default electron:electron_app
+gn gen out/ffmpeg "--args=import(\"//electron/build/args/ffmpeg.gn\") $GN_EXTRA_ARGS"
+
+if [ "$STRIP_BINARIES" == "true" ] && [ "`uname`" == "Linux" ]; then
+  if [ x"$TARGET_ARCH" == x ]; then
+    target_cpu=x64
+  elif [ "$TARGET_ARCH" == "ia32" ]; then
+    target_cpu=x86
+  else
+    target_cpu="$TARGET_ARCH"
+  fi
+  electron/script/copy-debug-symbols.py --target-cpu="$target_cpu" --out-dir=out/Default/debug -d out/Default --compress
+  electron/script/strip-binaries.py --target-cpu="$target_cpu" -d out/Default
+  electron/script/add-debug-link.py --target-cpu="$target_cpu" --debug-dir=out/Default/debug -d out/Default
+fi
+
+ninja -C out/ffmpeg electron:electron_ffmpeg_zip
+ninja -C out/Default electron:electron_dist_zip
+ninja -C out/Default electron:electron_mksnapshot_zip
+ninja -C out/Default electron:hunspell_dictionaries_zip
+ninja -C out/Default electron:electron_chromedriver_zip
+ninja -C out/Default third_party/electron_node:headers
+$SCCACHE_PATH --show-stats
+$SCCACHE_PATH --stop-server
+
+cd out
+# loop needed since sometimes the output directory is locked before completing the compilation
```

Comment:
is this true? i don't think linux or mac "lock" directories.

Reply:
I ported straight the code from the Windows part. What it was happening there was that sccache was getting lock of out directory and, until the server was stopped completely, the out directory was locked .  I think this one may even be removed, if we assume that --stop-server will always succeed.